src/main/java/com/gotocompany/depot/maxcompute/client/MaxComputeClient.java

```diff
@@ -0,0 +1,155 @@
+package com.gotocompany.depot.maxcompute.client;
+
+import com.aliyun.odps.Column;
+import com.aliyun.odps.Instance;
+import com.aliyun.odps.Odps;
+import com.aliyun.odps.OdpsException;
+import com.aliyun.odps.TableSchema;
+import com.aliyun.odps.account.Account;
+import com.aliyun.odps.account.AliyunAccount;
+import com.aliyun.odps.task.SQLTask;
+import com.aliyun.odps.tunnel.TableTunnel;
+import com.aliyun.odps.tunnel.TunnelException;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+import com.gotocompany.depot.maxcompute.client.insert.InsertManager;
+import com.gotocompany.depot.maxcompute.client.insert.InsertManagerFactory;
+import com.gotocompany.depot.maxcompute.exception.MaxComputeTableOperationException;
+import com.gotocompany.depot.maxcompute.model.RecordWrapper;
+import com.gotocompany.depot.maxcompute.schema.SchemaDifferenceUtils;
+import com.gotocompany.depot.metrics.Instrumentation;
+import com.gotocompany.depot.metrics.MaxComputeMetrics;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Slf4j
+public class MaxComputeClient {
+
+    private Odps odps;
+    private MaxComputeSinkConfig maxComputeSinkConfig;
+    private TableTunnel tableTunnel;
```

Comment:
The class is using multiple constructors with @AllArgsConstructor which makes object creation complex and error-prone. We should consider using the Builder pattern here instead. This would make the initialization flow much clearer and provide immutable fields. For example, we could have a private constructor that takes the essential dependencies and use @Builder to construct the object with proper validation. This would also help in unit testing as we can easily mock dependencies.